valhalla/sif/costfactory.h

```diff
@@ -22,15 +24,27 @@ namespace sif {
 /**
  * Generic factory class for creating objects based on type name.
  */
-template <class cost_t> class CostFactory {
+class CostFactory {
 public:
-  typedef std::shared_ptr<cost_t> cost_ptr_t;
-  typedef cost_ptr_t (*factory_function_t)(const Costing costing, const Options& options);
+  using factory_function_t = std::function<cost_ptr_t(const CostingOptions& options)>;
 
   /**
    * Constructor
    */
   CostFactory() {
```

Comment:
previously you had to call the deleted function below to register the built in costing models. since i was changing other stuff i figured it made sense to go ahead and remove that inconveneince